things/api/auth/http/endpoint.go

```diff
@@ -7,14 +7,15 @@ import (
 	"context"
 
 	"github.com/go-kit/kit/endpoint"
+	"github.com/mainflux/mainflux/pkg/errors"
 	"github.com/mainflux/mainflux/things"
 )
 
 func identifyEndpoint(svc things.Service) endpoint.Endpoint {
 	return func(ctx context.Context, request interface{}) (interface{}, error) {
 		req := request.(identifyReq)
 		if err := req.validate(); err != nil {
-			return nil, err
```

Comment:
Why did you change this?

Reply:
The default error being returned from validate function -> which is ErrBearerToken, causes the http response to return with 500 internal server error.
But, it should ideally return Statusunauthorized. That's why I changed this to return ErrAuthentication.

Reply:
In that case, it should be changed [here](https://github.com/mainflux/mainflux/blob/933b5dedce123063c6e2d17d0c6e842d2a203de9/things/api/auth/http/transport.go#L118).

Reply:
I made changes to transport.go, and reverted back to the original code in endpoints.go